scripts/production/repair_mlss_tags.py

```diff
@@ -0,0 +1,117 @@
+
+"""
+
+from argparse import ArgumentParser
+
+from ensembl.compara.db import DBConnection
+
+
+EXPECTED_VALUES_SQL = {
+    'max_align': """
+        SELECT method_link_species_set_id AS mlss_id, MAX(dnafrag_end - dnafrag_start) + 2 AS value
+        FROM constrained_element
+        GROUP BY method_link_species_set_id
+        UNION
+        SELECT method_link_species_set_id AS mlss_id, MAX(dnafrag_end - dnafrag_start) + 2 AS value
+        FROM genomic_align
+        GROUP BY method_link_species_set_id;
+    """,
+    'msa_mlss_id': """
+        SELECT mlss1.method_link_species_set_id AS mlss_id, mlss2.method_link_species_set_id AS value
+        FROM method_link_species_set mlss1
+        JOIN method_link_species_set mlss2 ON mlss1.species_set_id = mlss2.species_set_id
+        JOIN method_link ml1 ON mlss1.method_link_id = ml1.method_link_id
+        JOIN method_link ml2 ON mlss2.method_link_id = ml2.method_link_id
+        WHERE (ml1.class = "ConservationScore.conservation_score"
+            OR ml1.class = "ConstrainedElement.constrained_element")
+        AND (ml2.class = "GenomicAlignBlock.multiple_alignment" OR ml2.class LIKE "GenomicAlignTree.%%")
+        AND ml1.type NOT LIKE "pGERP%%"
+        AND ml2.type NOT LIKE "pEPO%%";
+    """
+}
+
+
+def repair_mlss_tag(dbc: DBConnection, mlss_tag: str) -> None:
+    """Repairs the given MLSS tag in the database, recomputing the expected values.
+
+    It uses the predefined query in `EXPECTED_VALUES_SQL` for the given MLSS tag to extract the
+    expected value for each MLSS id. It also adds any missing tags as well as removes any rows that
+    may contain an invalid MLSS id.
+
+    Args:
+        dbc: Compara database connection handler.
+        mlss_tag: MLSS tag as found in the ``method_link_species_set_tag`` table.
+
+    """
+    with dbc.connect() as connection:
+        # Get the MLSS tags in method_link_species_set_tag table
+        mlss_tag_values = connection.execute(f"SELECT method_link_species_set_id AS mlss_id, value "
+                                             f"FROM method_link_species_set_tag WHERE tag = '{mlss_tag}';")
+        mlss_tags = {row.mlss_id: row.value for row in mlss_tag_values.fetchall()}
+        # Extract the expected tag value based on the source data
+        expected_values = connection.execute(EXPECTED_VALUES_SQL[mlss_tag])
+        # Check that each tag has the correct value, fixing those that do not, and add any missing tags
+        for row in expected_values:
+            if row.mlss_id in mlss_tags:
+                # NOTE: due to internal conversions, we need to ensure both sides have the same time
+                if str(mlss_tags[row.mlss_id]) != str(row.value):
+                    connection.execute(f'UPDATE method_link_species_set_tag SET value = {row.value} '
+                                       f'WHERE method_link_species_set_id = {row.mlss_id} '
+                                       f'    AND tag = "{mlss_tag}";')
```

Comment:
```suggestion
                                       f'    AND tag = "{mlss_tag}"')
```
One more `;` - just in case you end up removing it 😉

Reply:
@JAlvarezJarreta has been very consistent in his `;` use - so I don't think it makes any difference removing it, unless he agrees to - and if he does, it would need removing from every query - otherwise he shouldn't change it at all, because consistency is key! :grin: